library/Icingadb/Model/Service.php

```diff
@@ -124,6 +132,31 @@ public function createBehaviors(Behaviors $behaviors)
         ]));
     }
 
+    /**
+     * Mutates flattened custom vars to an associative array
+     *
+     * @param $factory
+     *
+     * @return array
+     */
+    public function mutateVarsProperty($factory)
+    {
+        if (! $factory instanceof ResultSet) {
+            $query = $factory($this);
+            $this->applyRestrictions($query);
+        } else {
+            $query = $factory;
+        }
+
+        $vars = [];
+        foreach ($query as $customVar) {
+            $paths = explode('.', $customVar->flatname);
+            $this->transformVar($vars, $paths, $customVar->flatvalue);
```

Comment:
I thought of just using `$vars[$customVar->flatname] = $customVar->flatvalue` suffices. This also makes resolving macros easier, as it's just a single key lookup then.